src/cryptonote_core/service_node_list.cpp

```diff
@@ -723,11 +718,16 @@ namespace service_nodes
     // and if we don't already have the maximum
     auto contrib_iter = std::find_if(contributors.begin(), contributors.end(),
         [&address](const service_node_info::contribution& contributor) { return contributor.address == address; });
-    if (contrib_iter == contributors.end())
-    {
-      if (contributors.size() >= MAX_NUMBER_OF_CONTRIBUTORS || transferred < info.get_min_contribution())
-        return;
-    }
+
+    const bool contributer_found = (contrib_iter != contributors.end());
+
+    /// Only create a new contributor if we don't exceed the contributor limit
+    if (!contributer_found && contributors.size() >= MAX_NUMBER_OF_CONTRIBUTORS) return;
```

Comment:
Comment here is redundant with `contributors.size() >= MAX_NUMBER_OF_CONTRIBUTORS`.

Also this seems clearer if it uses the same branching as before and changing the boolean to be true by default, i.e.

```
const bool new_contributor = (contrib_iter == contributors.end());
if (new_contributor)
{
    if (contributors.size() >= MAX_NUMBER_OF_CONTRIBUTORS) return;
    
    const uint8_t hf_version = m_blockchain.get_hard_fork_version(block_height);
    const uint64_t min_contribution = get_min_node_contribution(hf_version, info.staking_requirement, info.total_reserved, contributors.size());
    if (transferred < min_contribution) return;
}
```

Because they're both related to being a new contributor.